app/code/Magento/Quote/Model/ShippingMethodManagement.php

```diff
@@ -335,6 +335,7 @@ private function getShippingMethods(Quote $quote, $address)
         if ($isCustomerGroupChanged) {
             $quote->setCustomerGroupId($quoteCustomerGroupId);
         }
+        $quote->save();
```

Comment:
Hi @kishorekumarkesavan. Thank you for your collaboration. We need to find another way for this solution. The quote must not be saved in `getShippingMethods` operation somehow since this operation must not have side-effects on quote in the database directly. 
It's a resource-consuming operation. If we need to save a quote with some new values, it should be performed on another level but not in the shipping methods collector method.